src/Stratis.Bitcoin.Features.BlockStore/BlockStoreSignaled.cs

```diff
@@ -13,10 +16,10 @@ namespace Stratis.Bitcoin.Features.BlockStore
 {
     public class BlockStoreSignaled : SignalObserver<Block>
     {
-        private readonly IAsyncLoopFactory asyncLoopFactory;
+        /// <summary>Maximum time in milliseconds for forming a new batch.</summary>
+        private const int FlushFrequencyMilliseconds = 500;
```

Comment:
i think 500 ms is too low, i'd go for 5 second

Reply:
Agreed. But only after we have a block store cache and we won't need to skip 1 iteration of announcing some blocks when we're trying to announce something we don't have on block store yet.

Reply:
I don't understand. I think 5 second "timeout" would be handy even without block store cache. Block store cache only allows us to merge cases 1 and 2, the batch case is case 3, so the cache does not affect that.

Reply:
What if we just received a block and mined another one on top of that 1 sec after? We will end up waiting 4 sec before it gets propagated.

Reply:
no, see the alg i just posted